File: Clojure/Planner/src/App/schema.clj
(ns App.schema)

(defrecord Activity [id-location ^double value])
(defrecord Location [^double lat ^double lon])
(defrecord Resource [id-location ^double cost-ph ^double cost-ph-ot ^double cost-km])

(defn new-schedule []
  { :activity {} :location {} :resource {} })

File: C++/Scheduler/schema/schema.cpp
#include <map>
#include "activity.hpp"
#include "allocation.hpp"
#include "location.hpp"

class Schema {
public:
    std::map<std::string, Activity> activity;
    std::map<std::string, Allocation> allocation;
    std::map<std::string, Location> location;
};